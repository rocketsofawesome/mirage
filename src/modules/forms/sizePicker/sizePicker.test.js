import React from 'react'
import 'jest-styled-components'
import { SquareRadioButton, SizePicker } from 'SRC'

const { mountWithTheme } = global

const defaultProps = {
  productId: '12345',
  currentSize: 'sku-12345',
  onSizeSelect: () => {},
  variants: [
    {sku: "sku-12345", size: 3, inStock: true},
    {sku: "sku-21354", size: 4, inStock: true},
    {sku: "sku-13245", size: 5, inStock: true},
    {sku: "sku-15234", size: 6, inStock: true},
    {sku: "sku-14235", size: 7, inStock: false},
    {sku: "sku-13254", size: 9, inStock: true},
    {sku: "sku-13534", size: 12, inStock: true},
  ]
}

describe('() SizePicker', () => {
  const createSizePicker = (inProps) => {
    const props = {
      ...defaultProps,
      ...inProps
    }
    return mountWithTheme(<SizePicker {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createSizePicker())
    .toMatchSnapshot()
  })

  test('renders out proper amount of radio buttons', () => {
    expect(createSizePicker().find(SquareRadioButton).length).toEqual(defaultProps.variants.length)
  })

  test('current size is being selected', () => {
    expect(
      createSizePicker()
      .find({ selected: true })
      .first()
      .prop('input')
      .value
    ).toEqual(defaultProps.currentSize)
  })
})