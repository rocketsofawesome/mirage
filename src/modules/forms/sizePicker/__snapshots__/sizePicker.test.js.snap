// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`() SizePicker matching the snapshot 1`] = `
.c2 {
  box-sizing: border-box;
  width: 6rem;
  height: 4.7rem;
  border-radius: 0.2rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  margin-bottom: 0;
  font-family: "ff-din-web",Helvetica,Arial,sans-serif;
  font-size: 1.4rem;
  font-weight: 200;
  cursor: pointer;
  color: #00003C;
  border: solid 0.1rem #0073D1;
  position: relative;
}

.c2:hover {
  background-color: #BCE1F7;
}

.c2 > input {
  position: absolute;
  height: 0;
  width: 0;
  opacity: 0;
}

.c2.disabled {
  border: solid 0.1rem #979797;
  color: #979797;
}

.c2.disabled:hover {
  background-color: transparent;
  cursor: not-allowed;
}

.c2.selected {
  background-color: #BCE1F7;
}

.c0 .c1 {
  margin-top: 1.5rem;
  margin-right: 0.75rem;
  margin-left: 0.75rem;
}

<Styled(Component)
  currentSize="sku-12345"
  productId="12345"
  variants={
    Array [
      Object {
        "inStock": true,
        "size": 3,
        "sku": "sku-12345",
      },
      Object {
        "inStock": true,
        "size": 4,
        "sku": "sku-21354",
      },
      Object {
        "inStock": true,
        "size": 5,
        "sku": "sku-13245",
      },
      Object {
        "inStock": true,
        "size": 6,
        "sku": "sku-15234",
      },
      Object {
        "inStock": false,
        "size": 7,
        "sku": "sku-14235",
      },
      Object {
        "inStock": true,
        "size": 9,
        "sku": "sku-13254",
      },
      Object {
        "inStock": true,
        "size": 12,
        "sku": "sku-13534",
      },
    ]
  }
>
  <Component
    className="c0"
    currentSize="sku-12345"
    productId="12345"
    variants={
      Array [
        Object {
          "inStock": true,
          "size": 3,
          "sku": "sku-12345",
        },
        Object {
          "inStock": true,
          "size": 4,
          "sku": "sku-21354",
        },
        Object {
          "inStock": true,
          "size": 5,
          "sku": "sku-13245",
        },
        Object {
          "inStock": true,
          "size": 6,
          "sku": "sku-15234",
        },
        Object {
          "inStock": false,
          "size": 7,
          "sku": "sku-14235",
        },
        Object {
          "inStock": true,
          "size": 9,
          "sku": "sku-13254",
        },
        Object {
          "inStock": true,
          "size": 12,
          "sku": "sku-13534",
        },
      ]
    }
  >
    <div
      className="c0"
    >
      <Styled(Component)
        disabled={true}
        input={
          Object {
            "checked": true,
            "disabled": undefined,
            "onChange": undefined,
            "value": "sku-12345",
          }
        }
        key="12345-0"
        selected={true}
      >
        <Component
          className="c1 c2"
          disabled={true}
          input={
            Object {
              "checked": true,
              "disabled": undefined,
              "onChange": undefined,
              "value": "sku-12345",
            }
          }
          selected={true}
        >
          <label
            className="c1 c2 selected disabled"
          >
            3
            <input
              checked={true}
              type="radio"
              value="sku-12345"
            />
          </label>
        </Component>
      </Styled(Component)>
      <Styled(Component)
        disabled={true}
        input={
          Object {
            "checked": false,
            "disabled": undefined,
            "onChange": undefined,
            "value": "sku-21354",
          }
        }
        key="12345-1"
        selected={false}
      >
        <Component
          className="c1 c2"
          disabled={true}
          input={
            Object {
              "checked": false,
              "disabled": undefined,
              "onChange": undefined,
              "value": "sku-21354",
            }
          }
          selected={false}
        >
          <label
            className="c1 c2 disabled"
          >
            4
            <input
              checked={false}
              type="radio"
              value="sku-21354"
            />
          </label>
        </Component>
      </Styled(Component)>
      <Styled(Component)
        disabled={true}
        input={
          Object {
            "checked": false,
            "disabled": undefined,
            "onChange": undefined,
            "value": "sku-13245",
          }
        }
        key="12345-2"
        selected={false}
      >
        <Component
          className="c1 c2"
          disabled={true}
          input={
            Object {
              "checked": false,
              "disabled": undefined,
              "onChange": undefined,
              "value": "sku-13245",
            }
          }
          selected={false}
        >
          <label
            className="c1 c2 disabled"
          >
            5
            <input
              checked={false}
              type="radio"
              value="sku-13245"
            />
          </label>
        </Component>
      </Styled(Component)>
      <Styled(Component)
        disabled={true}
        input={
          Object {
            "checked": false,
            "disabled": undefined,
            "onChange": undefined,
            "value": "sku-15234",
          }
        }
        key="12345-3"
        selected={false}
      >
        <Component
          className="c1 c2"
          disabled={true}
          input={
            Object {
              "checked": false,
              "disabled": undefined,
              "onChange": undefined,
              "value": "sku-15234",
            }
          }
          selected={false}
        >
          <label
            className="c1 c2 disabled"
          >
            6
            <input
              checked={false}
              type="radio"
              value="sku-15234"
            />
          </label>
        </Component>
      </Styled(Component)>
      <Styled(Component)
        disabled={true}
        input={
          Object {
            "checked": false,
            "disabled": undefined,
            "onChange": undefined,
            "value": "sku-14235",
          }
        }
        key="12345-4"
        selected={false}
      >
        <Component
          className="c1 c2"
          disabled={true}
          input={
            Object {
              "checked": false,
              "disabled": undefined,
              "onChange": undefined,
              "value": "sku-14235",
            }
          }
          selected={false}
        >
          <label
            className="c1 c2 disabled"
          >
            7
            <input
              checked={false}
              type="radio"
              value="sku-14235"
            />
          </label>
        </Component>
      </Styled(Component)>
      <Styled(Component)
        disabled={true}
        input={
          Object {
            "checked": false,
            "disabled": undefined,
            "onChange": undefined,
            "value": "sku-13254",
          }
        }
        key="12345-5"
        selected={false}
      >
        <Component
          className="c1 c2"
          disabled={true}
          input={
            Object {
              "checked": false,
              "disabled": undefined,
              "onChange": undefined,
              "value": "sku-13254",
            }
          }
          selected={false}
        >
          <label
            className="c1 c2 disabled"
          >
            9
            <input
              checked={false}
              type="radio"
              value="sku-13254"
            />
          </label>
        </Component>
      </Styled(Component)>
      <Styled(Component)
        disabled={true}
        input={
          Object {
            "checked": false,
            "disabled": undefined,
            "onChange": undefined,
            "value": "sku-13534",
          }
        }
        key="12345-6"
        selected={false}
      >
        <Component
          className="c1 c2"
          disabled={true}
          input={
            Object {
              "checked": false,
              "disabled": undefined,
              "onChange": undefined,
              "value": "sku-13534",
            }
          }
          selected={false}
        >
          <label
            className="c1 c2 disabled"
          >
            12
            <input
              checked={false}
              type="radio"
              value="sku-13534"
            />
          </label>
        </Component>
      </Styled(Component)>
    </div>
  </Component>
</Styled(Component)>
`;
